src/pulses/gaussian.jl

```diff
@@ -0,0 +1,54 @@
+
+############################
+# Gaussian pulsed plane wave
+############################
+
+"""
+    GaussianPulse(mom::M,pulse_width::T) where {M<:QEDbase.AbstractFourMomentum,T<:Real}
+
+Concrete implementation of an `AbstractPulsedPlaneWaveField` for Gaussian pulses.
+
+Propagates along the direction given by the space-like components of the reference momentum vector k_mu = (omega/speed_of_light, k_x, k_y, k_z), and omega = 2*pi*speed_of_light/wavelength.
+
+The time-like coordinate omega/speed_of_ligth of k_mu defines the field's oscillation frequency.
+
+In order to fulfill the vacuum dispersion relation, k_mu*k^mu=0 is required.
+
+!!! note "Pulse shape"
+
+    The longitudinal envelope of a Gaussian pulse is defined as
+
+    ```math
+    g(\\phi) = \\exp(-\\frac{\\phi^2}{\\Delta\\phi^2})
+    ```
+    for \$\\phi\\in (-7\\Delta\\phi, 7\\Delta\\phi),\$where \$\\Delta\\phi`\$ denotes the `pulse_width`, and zero otherwise.
+
+    There is no envelope in the transverse directions.
+
+"""
+struct GaussianPulse{M<:QEDbase.AbstractFourMomentum,T<:Real} <:
+       AbstractPulsedPlaneWaveField
+    mom::M
+    pulse_width::T
+end
+
+@inline function _unsafe_gaussian_envelope(phi, dphi)
+    return exp(-(phi / dphi)^2)
```

Comment:
I think there should be a two in the denominator. Then the `dphi^2` is identical to the variance of the Gaussian.

Reply:
Just a matter of taste...But will do.